src/radical/pilot/task.py

```diff
@@ -490,11 +560,48 @@ def wait(self, state=None, timeout=None):
     # --------------------------------------------------------------------------
     #
     def cancel(self):
-        """
+        '''
         Cancel the task.
-        """
+        '''
 
         self._tmgr.cancel_tasks(self.uid)
 
 
 # ------------------------------------------------------------------------------
+#
+class TaskDict(ru.TypedDict):
```

Comment:
is this something for future consideration? should it be a separate module (and to keep in a dedicated level branch)?

Reply:
this works (after our discussion and thinking more - I got a better perspective on it, this is a right thing to have)

p.s. a little bit off topic - do we want to have a particular approach in naming for such data (kinda to reflect its purpose - keep state and processing info)? -> `TaskInfo` (similar to `RMInfo`)

Reply:
Naming:  they are a bit different in their role: `TaskDict` is a dictionary representation of a task object, `RMInfo` are information collect by the Rresource Manager.  So in those specific instances I would stick with the current names.